code/modules/mob/living/carbon/xenomorph/XenoProcs.dm

```diff
@@ -700,6 +700,63 @@
 		tracked_marker.xenos_tracking -= src
 	tracked_marker = null
 
+/mob/living/carbon/Xenomorph/proc/do_nesting_host(mob/M, nest_structural_base)
+	var/list/xeno_hands = list(get_active_hand(), get_inactive_hand())
+
+	if(!ishuman(M))
+		to_chat(src, SPAN_XENONOTICE("This is not a host."))
+		return
+
+	var/mob/living/carbon/human/H = M
+
+	for(var/i; i != 0; i = length(xeno_hands))
+		if(isnull(xeno_hands[i]))
+			xeno_hands -= xeno_hands[i]
+		else if(istype(xeno_hands[i], /obj/item/grab))
+			break
+
+	if(!length(xeno_hands))
+		to_chat(src, SPAN_XENONOTICE("You must have a surer hold on the host to nest them!"))
+		return
+
+	var/turf/supplier_turf = get_turf(nest_structural_base)
+	var/obj/effect/alien/weeds/supplier_weeds = locate(/obj/effect/alien/weeds) in supplier_turf
+	if(!supplier_weeds)
+		to_chat(src, SPAN_XENOBOLDNOTICE("There are no weeds here! Nesting hosts requires hive weeds."))
+		return
+
+	if(supplier_weeds.weed_strength < WEED_LEVEL_HIVE)
+		to_chat(src, SPAN_XENOBOLDNOTICE("The weeds here are not strong enough for nesting hosts."))
+		return
+
+	if(!supplier_turf.density)
+		var/obj/structure/window/framed/framed_window = locate(/obj/structure/window/framed/) in supplier_turf
+		if(!framed_window)
+			to_chat(src, SPAN_XENOBOLDNOTICE("Hosts need a vertical surface to be nested upon!"))
+			return
+
+	var/dir_to_nest = get_dir(H, nest_structural_base)
+
+	if(!locate(dir_to_nest) in GLOB.cardinals)
+		to_chat(src, SPAN_XENONOTICE("The host must be directly next to the wall its being nested on!"))
+		return
+
+	for(var/obj/structure/bed/nest/N in supplier_weeds.nesting_sites)
+		if(get_dir(nest_structural_base, H) == get_dir(supplier_weeds, N))
+			to_chat(src, SPAN_XENOBOLDNOTICE("Theres already a host nested here!"))
+			return
+
+	var/obj/structure/bed/nest/funny_nest = new(get_turf(H))
+	supplier_weeds.nesting_sites |= funny_nest
+	funny_nest.foundation = supplier_weeds
+	funny_nest.dir = dir_to_nest
+	if(!funny_nest.buckle_mob(H, src))
+		supplier_weeds.nesting_sites -= funny_nest
+		qdel(funny_nest)
+
```

Comment:
Reminder: change this to alter the host’s layer and not the walls, so other nests on the wall dont get obscured